### DISCUSSION TITLE
Is there a way increase the number of characters in an event log?

### DISCUSSION THREAD

**Discussion by dheadrick1618** [nasa/fprime#2010] [Q&A] [answered]
I am running the RPI demo project on a raspberry pi 3B right now, and sometimes it appears to crash for no reason giving me errors suggesting the GPIO, LED, and UART drivers cannot be opened. I am trying to debug this , but the error event provided gives me a path to the file is **cut short** by an apparent character limit on the event log message size. The following is the event log output I am trying to work with for reference: 

fatalAdapter.AF_ASSERT_1,1001,EventSeverity.FATAL,Assert in file /home/fprime-dev/fprime_rpi_test/rpi_test/fprime/RPI/build-fprime-automatic-arm-, line 2866: 8

You can see that the path ends after 'build-fprime-automatic-arm- ', but this doesn't help me much and in the future I worry I will come across issues where I need this path to troubleshoot different errors. 

Please let me know if there is a work around, or if I can tweak a setting to allow the event log to print the entire path regardless of its character length.

**Comment by LeStarch** [comment, marked as answer]
I find the easiest way to solve this is to set the [configuration](https://github.com/nasa/fprime/blob/5c05e60136daf601c8320d9ab3e7d07d47e3f895/config/FpConfig.h#L163) to use `FW_RELATIVE_PATH_ASSERT` instead of `FW_FILENAME_ASSERT`.  This throws away all the path from matter (/home/.....) and displays a relative path in repository.  That way you can see more.

To answer your specific question on increasing that size, you'll likely need to increase several properties:
- [FW_ASSERT_DFL_MSG_LEN](https://github.com/nasa/fprime/blob/5c05e60136daf601c8320d9ab3e7d07d47e3f895/config/FpConfig.h#LL161C9-L161C30): assert message length
- [FW_ASSERT_TEXT_SIZE](https://github.com/nasa/fprime/blob/5c05e60136daf601c8320d9ab3e7d07d47e3f895/config/FpConfig.h#LL168C9-L168C28): assert text size
- [FW_COM_BUFFER_MAX_SIZE](https://github.com/nasa/fprime/blob/master/config/FpConfig.h#LL168C9-L168C28): com buffer size, used to carry all events thus needs to be big enough for event arguments.
- [FW_LOG_STRING_MAX_SIZE](https://github.com/nasa/fprime/blob/master/config/FpConfig.h#LL168C9-L168C28): log string max size (max string for events)
- [FW_LOG_TEXT_BUFFER_SIZE](https://github.com/nasa/fprime/blob/master/config/FpConfig.h#LL168C9-L168C28): buffer size for text-represented  events
- Event string sizes for assert fatal adapter: [e.g.](https://github.com/nasa/fprime/blob/5c05e60136daf601c8320d9ab3e7d07d47e3f895/Svc/AssertFatalAdapter/AssertFatalEvents.fppi#L12): local component string sizes

Hence why it might be easier to uses the truncated version mentioned first.

**Reply by dheadrick1618** [reply]
Thank you. For now I will rely on gdb and your truncated method for debugging. Is there any particular reason the assert text size is limited to 120, while the string buffer size is 255?
